src/descriptions/id-broker-api.php

```diff
@@ -11,12 +11,17 @@
             'responseModel' => 'Result',
             'parameters' => [
                 'username' => [
-                    'required' => true,
+                    'required' => false,
```

Comment:
It looks like `authenticateNewUserInternal` was partially combined with `authenticateInternal` (e.g. an `@method` for `authenticateNewUserInternal()` was added to [IdBrokerClient.php](https://github.com/silinternational/idp-id-broker-php-client/pull/46/files#diff-023d0d33cffa6f8505f05501be0120afR12), but only `authenticateInternal()` seems to be defined/used.

I think it would be good to keep them as separate methods, with separate definitions here in this file. Or was there some problem that made that impractical?

Reply:
I went through a couple iterations before I settled on this implementation. It looks like I left the @method by accident. I decided it made more sense to have separate wrapper methods for the single API endpoint since it's two clearly unique use cases. Didn't really run into any problems with the alternative implementation that you describe. It was a judgement call. Thoughts?

Reply:
Since a normal authentication requires different parameters than a new-user authentication, it seems worth having even a separate entry here (authenticateNewUserInternal) so that we can be clear about what parameters are required for which type of authentication.